From: Jan Engelhardt <jengelh@inai.de>
Date: 2024-10-28 10:10:20.918922623 +0100
References: https://github.com/protocolbuffers/protobuf/pull/19009

Unversioned libraries are strongly discouraged. Use
https://en.opensuse.org/openSUSE:Shared_library_packaging_policy#When_there_is_no_versioning
method 1 to remedy. Though utf8_range has a version of its own ("1.0"
visible through the .pc file) and gets third_party/-like treatment,
protobuf is the authoritative repository for it, using the protobuf
version for our SONAME seems acceptable.

This openSUSE patch follows SLPP's naming provisions and so is
slightly different from PR19009 while the PR is unmerged.

---
 third_party/utf8_range/CMakeLists.txt |    8 ++++++++
 1 file changed, 8 insertions(+)

Index: protobuf-29.3/third_party/utf8_range/CMakeLists.txt
===================================================================
--- protobuf-29.3.orig/third_party/utf8_range/CMakeLists.txt
+++ protobuf-29.3/third_party/utf8_range/CMakeLists.txt
@@ -19,6 +19,9 @@ add_library (utf8_range
 # A heavier-weight C++ wrapper that supports Abseil.
 add_library (utf8_validity utf8_validity.cc utf8_range.c)
 
+set_target_properties(utf8_range PROPERTIES OUTPUT_NAME ${LIB_PREFIX}utf8_range-${protobuf_VERSION})
+set_target_properties(utf8_validity PROPERTIES OUTPUT_NAME ${LIB_PREFIX}utf8_validity-${protobuf_VERSION})
+
 # Load Abseil dependency.
 if (NOT TARGET absl::strings)
   if (NOT ABSL_ROOT_DIR)
